packages/react/src/components/Button/Button.tsx

```diff
@@ -193,6 +193,14 @@ const Button: ButtonComponent = React.forwardRef(
       }
 
       return (
+        // @ts-expect-error - `IconButton` does not support all `size`s that
+        // `Button` supports.
+        //
+        // TODO: What should be done here?
+        // 1. Should the `IconButton` not be rendered if the `size` is not
+        //    supported?
+        // 2. Should an error be thrown?
+        // 3. Something else?
```

Comment:
> What should be done here?

To avoid a breaking change, nothing. To help encourage best practices we could update the typing to be a mix of what you have and what we had before. In other components we define a "deprecated" type and then union it with the non-deprecated types. `align` in Popover does this as an example

In the next major the deprecated ones would go away and the types would limit it to only the supported sizes.